pilot/pkg/model/context.go

```diff
@@ -308,12 +309,33 @@ func (s *StringBool) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// ProxyConfig can only be marshalled using (gogo) jsonpb. However, the rest of node meta is not a proto
+// To allow marshalling, we need to define a custom type that calls out to the gogo marshaller
+type JsonProxyConfig meshconfig.ProxyConfig
+
+func (s JsonProxyConfig) MarshalJSON() ([]byte, error) {
+	var buf bytes.Buffer
+	pc := meshconfig.ProxyConfig(s)
+	if err := (&gogojsonpb.Marshaler{}).Marshal(&buf, &pc); err != nil {
+		return nil, err
+	}
+	return buf.Bytes(), nil
+}
+
+func (s *JsonProxyConfig) UnmarshalJSON(data []byte) error {
+	pc := (*meshconfig.ProxyConfig)(s)
+	return gogojsonpb.Unmarshal(bytes.NewReader(data), pc)
+}
+
 // NodeMetadata defines the metadata associated with a proxy
 // Fields should not be assumed to exist on the proxy, especially newly added fields which will not exist
 // on older versions.
 // The JSON field names should never change, as they are needed for backward compatibility with older proxies
 // nolint: maligned
 type NodeMetadata struct {
+	// ProxyConfig defines the proxy config specified for a proxy.
+	ProxyConfig *JsonProxyConfig `json:"PROXY_CONFIG,omitempty"`
```

Comment:
That's super ugly - json in a struct ? With gogo mixed with go proto ?
And NodeMeta is sent quite often - while ProxyConfig keeps growing in size.

Reply:
The full node meta is 19778 bytes on a pod I have. Without ProxyConfig its 19036, so this is only about a 5% increase. I think its worth it. Note in the future we only send this on the first request, but its broken due to a Envoy bug unfortunately.

Reply:
Its not raw json, maybe the naming is confusing. We just need a custom type name (I think) that aliases Proxyconfig so we can define custom json marshalling logic for it